packages/NodeCollector/BinaryOpTreeRootLocator.php

```diff
@@ -20,20 +24,32 @@ final class BinaryOpTreeRootLocator
      */
     public function findOperationRoot(Expr $expr, string $binaryOpClass): Expr
     {
-        /** @var ?Expr $parentNode */
         $parentNode = $expr->getAttribute(AttributeKey::PARENT_NODE);
-        if ($parentNode === null || $parentNode::class !== $binaryOpClass) {
+        if (! $parentNode instanceof Node) {
```

Comment:
As mentioned in https://github.com/rectorphp/rector-src/pull/1621#discussion_r778089622, combined with the next check, this is just a less clear synonym for `$parentNode === null`.